LogicLayer/Notifications/MailSender.cs

```diff
@@ -0,0 +1,115 @@
+
+        public MailSender(IOptions<MailNotificationsConfiguration> mailOptionsAccessor)
+        {
+            MailOptions = mailOptionsAccessor.Value;
+        }
+
+        public Task SendEmailConfirmationAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EmailConfirmationHtml),
+                link,
+                "Email Confirmation");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendEventCancellationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventCancellationHtml),
+                link,
+                "Event Cancellation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventConfirmationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventConfirmationHtml),
+                link,
+                "Event Confirmation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventInvitationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.EventInvitationHtml), link, "Event Invitation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendFriendAcceptAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendAcceptHtml),
+                link,
+                "Friend Invitation Accepted");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendFriendInvitationAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendInvitationHtml),
+                link,
+                "Friend Invitation");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendPasswordResetAsync(string email, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.PasswordResetHtml), link, "Password Reset");
+            return SendEmailAsync(email, notification);
+        }
+
+        private Task SendEmailAsync(IEnumerable<string> emails, Notification notification)
+            => Execute(MailOptions.SendGridApiKey, emails, notification);
+
+        private Task SendEmailAsync(string email, Notification notification)
+        {
+            var emails = new List<string>
+            {
+                email
+            };
+            return SendEmailAsync(emails, notification);
+        }
+
+        private Task Execute(string apiKey, IEnumerable<string> emails, Notification notification)
```

Comment:
Again, listen to R# - this can be `static`.
Additionally, async methods should be named with an `Async` suffix, so `ExecuteAsync`.

Reply:
Oh, and `Execute` is a terrible name, I have no idea what is actually executed before I read this method's contents, change the name to something meaningful.

Reply:
Done